lamindb/core/storage/_tiledbsoma.py

```diff
@@ -192,5 +188,10 @@ def save_tiledbsoma_experiment(
         )
 
     return Artifact(
-        storepath, key=key, description=description, run=run, revises=revises, _uid=_uid
+        storepath,
+        key=key,
+        description=description,
+        run=run,
+        revises=revises,
+        _is_internal=True,
```

Comment:
FYI @Koncopd 

I'm no longer passing `_uid` here because I'm worried about inconsistencies between the uid that's encoded in the storepath and the passed variable.

Instead, I'm now parsing the `uid` from the storepath within the Artifact constructor.

The `_is_internal` variable wouldn't be needed and is just for raising clearer errors to users who should never pass a path or key that contains `.lamindb`.

Reply:
I see, makes perfect sense.

Reply:
Also a similar thing happens here https://github.com/laminlabs/lamindb/blob/4d6273cc5c9ea4ac5a6f7ae5c4e73ee26a5bc034/lamindb/_artifact.py#L949C44-L949C52
Does this also need an edit?

Reply:
The link is broken but I guess you mean this:

https://github.com/laminlabs/lamindb/blob/4d6273cc5c9ea4ac5a6f7ae5c4e73ee26a5bc034/lamindb/_artifact.py#L949

This looks good because it doesn't pass a uid!

Reply:
Oh, yes, it now raises this error because the constructor thinks it's coming from a user.

```
>               raise ValueError("Do not pass paths inside the `.lamindb` directory.")
E               ValueError: Do not pass paths inside the `.lamindb` directory.
```

So, I'll pass `_is_internal_call`.

Reply:
The problem is that it can actually come from a user.

Reply:
The path here can be both autogenerated and normal user defined.

Reply:
I don't think it's a problem because `_is_internal_call` only tells the API that it's OK to pass a `.lamindb/` path. Because the `.open()` API doesn't allow to pass an external path from the user, we should be good.

<img width="591" alt="image" src="https://github.com/user-attachments/assets/76118726-7511-4e09-9088-68eadc3def0e">